app/src/org/commcare/android/database/HybridFileBackedStorageIterator.java

```diff
@@ -0,0 +1,46 @@
+package org.commcare.android.database;
+
+import android.database.Cursor;
+
+import org.commcare.modern.database.DatabaseHelper;
+import org.javarosa.core.services.storage.Persistable;
+
+import java.io.InputStream;
+
+/**
+ * Iterator for storage layer that uses filesystem to store payload.
+ *
+ * @author Phillip Mates (pmates@gmail.com)
+ */
+public class HybridFileBackedStorageIterator<T extends Persistable>
+        extends SqlStorageIterator<T> {
+
+    /**
+     * Creates an iterator into either a full or partial sql storage.
+     *
+     * @param c       The uninitialized cursor for a query.
+     * @param storage The storage being queried
+     */
+    public HybridFileBackedStorageIterator(Cursor c,
+                                           HybridFileBackedSqlStorage<T> storage) {
+        super(c, storage, null);
+    }
+
+    @Override
+    public T nextRecord() {
+        byte[] blob =
```

Comment:
Neither does the normal SqlIterator. I think the onus is currently on the caller to invoke this correctly.